add-on/chrome/modules/LoopRooms.jsm

```diff
@@ -746,9 +746,26 @@ var LoopRoomsInternal = {
       eventEmitter.emit("close");
     });*/
 
-    let win = Services.wm.getMostRecentWindow("navigator:browser");
-    win.LoopUI.loadSidebar(roomToken);
-    MozLoopService.openURL(TOC_URL + roomToken);
+    let urlToLoad =
+      Cc["@mozilla.org/supports-string;1"].createInstance(Ci.nsISupportsString);
+    urlToLoad.data = TOC_URL + roomToken;
+
+    let windowArgs =
+      Cc["@mozilla.org/supports-array;1"].createInstance(Ci.nsISupportsArray);
+    windowArgs.AppendElement(urlToLoad);
+
+    // If we ever decide we want to do anything slightly more sophisticated (eg
+    // allow Hello in private browsing mode), we may want to use
+    // utilityOverlay.js:openUILinkIn (which calls
+    // utilityOverlay.js:openLinkIn).  Doing this is rather gross, as we would
+    // need to get the most recently used window and then reach in and invoke
+    // window.openUILinkIn, and it's also conceivably fragile, since this is
+    // unlikely to be a supported way to use that API.
```

Comment:
I think I'd like to see the alternative that I suggested added here as well. Something like "Alternately, we set an object and token on the window object for the newly created window, and use that as our flag".